Ft data-modal

zbahati opened the pull request:
- Add files
- Create application migrations 
- A comment model 
- Add like model 
- Add post model
- Add user model 
- Add relationship between them using ERD schema 
- Add indexes for the foreign keys created for search optimization


# Finishes ...

zbahati committed:
Ft data-modal
- Add files
- Create application migrations
- Add models class for the migrations
- already finish two tables users table , and posts tables with authorid refrence to the post

Commit:
- Add comments model
-Add like model
- migrate to make sure that rails server is running without error

Commit:
- Add rubocop for ruby codes espect
 - Add styles linters for the css styles
- Remeove all linters error

# Finishes ..

Commit:
Correct the readme file was empty now i populate it with the description

omar-muhamad commented on db/migrate/20231025144856_create_users.rb:
```diff
@@ -0,0 +1,12 @@
+      t.string :Name
+      t.text :Photo
+      t.string :Bio
+      t.integer :PostsCounter
+
```
- [ ] You should use snake_case for naming database columns according to the [Rails naming conventions](https://gist.github.com/iangreenleaf/b206d09c587e8fc6399e#database). Please name this column `:posts_counter`, instead of `:PostsCounter`.

omar-muhamad commented on db/migrate/20231025151859_create_posts.rb:
```diff
@@ -0,0 +1,13 @@
+      t.references :AuthorId, foreign_key: { to_table: :users }
+      t.string :Title
+      t.text :Text
+      t.integer :CommentsCounter
+      t.integer :LikesCounter
```
- [ ] Same here, You should use snake_case for naming database columns according to the [Rails naming conventions](https://gist.github.com/iangreenleaf/b206d09c587e8fc6399e#database). Please name these columns `:comments_counter` and `:likes_counter`, instead of `:CommentsCounter` and `:LikesCounter`.

omar-muhamad commented on db/migrate/20231025151859_create_posts.rb:
```diff
@@ -0,0 +1,13 @@
+class CreatePosts < ActiveRecord::Migration[7.1]
+  def change
+    create_table :posts do |t|
+      t.references :AuthorId, foreign_key: { to_table: :users }
+      t.string :Title
```
- [ ] As per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/creating_data_model_project_v1_2.md#:~:text=should%20be%20included.-,Pay%20attention%20that%20in%20posts%20table%20the%20column%20holding%20the%20foreign%20key%20to%20the%20users%20table%20should%20be%20named%20author_id,-All%20columns%20that), The foreign key between users and posts table is `author_id`, not `AuthorId_id` as you did here. Please fix that.

See how it's created in the schema file:

![image](https://github.com/zbahati/blogApp/assets/68796232/640aaa79-1f52-4786-a807-4d3d46b6d23c)


[Hint] It will be like that:
```ruby
t.references :author, foreign_key: { to_table: 'users' } 
```

omar-muhamad commented on db/migrate/20231026063935_create_likes.rb:
```diff
@@ -0,0 +1,10 @@
+class CreateLikes < ActiveRecord::Migration[7.1]
+  def change
+    create_table :likes do |t|
+      t.references :userID, foreign_key: {to_table: :users}
+      t.references :postId, foreign_key: {to_table: :posts}
```
- [ ] The foreign key between users and likes table is `user_id`, not `userId_id`. And between posts and likes tables is `post_id`, not `postId_id` as you did here. Please fix that.

See how it's created in the schema file:

![image](https://github.com/zbahati/blogApp/assets/68796232/bec5a5e5-0778-4f8b-97cc-e19b006c256f)

[Hint] It will be like that:
```ruby
 t.references :user, foreign_key: true 
 t.references :post, foreign_key: true
```

omar-muhamad commented on db/migrate/20231026064911_create_comments.rb:
```diff
@@ -0,0 +1,11 @@
+class CreateComments < ActiveRecord::Migration[7.1]
+  def change
+    create_table :comments do |t|
+      t.references :UserId, foreign_key: {to_table: :users}
+      t.references :PostId, foreign_key: {to_table: :posts}
```
- [ ] Same here, The foreign key between users and comments tables is `user_id`, not `userId_id`. And between posts and comments tables is `post_id`, not `postId_id` as you did here. Please fix that.

See how it's created in the schema file:

![image](https://github.com/zbahati/blogApp/assets/68796232/bb525704-a7a6-4a79-9f0c-5518e5cf2026)

[Hint] It will be like that:
```ruby
 t.references :user, foreign_key: true 
 t.references :post, foreign_key: true
```

omar-muhamad reviewed (changes requested):
# Changes Required ♻️
### Hi @zbahati,

### Nice job, you are almost there.

![cat-driving-serious](https://user-images.githubusercontent.com/68796232/223284600-852bcd90-1495-4a40-8515-cc12636b369b.gif)

## (Highlights) Good Points: :+1:
- Your work is well documented. ✔
- The correct workflow is used.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please check the inline comments below ⬇️.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

Commit:
Add changes for the CammelCase to snake Case and other changes

- Add migration file to change Column of Likes
- Add migration file to change columns of comments
- Add migration file to change columns of Posts
- Add migratio file to changes columns of Users

mikethreels reviewed (approved):
Hi Bahati, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Good job implementing the requested changes.
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
